tests/unit/algos/list/delimiterVersions.spec.js

```diff
@@ -154,13 +167,11 @@ const receivedData = [
     { key: 'notes/summer/1.txt', value: bar, versionId: 'bar' },
     { key: 'notes/summer/1.txt', value: foo, versionId: 'foo' },
     { key: 'notes/summer/2.txt', value: bar, versionId: 'bar' },
-    { key: 'notes/summer/4.txt', value: valueDeleteMarker, versionId: 'bar' },
-    { key: 'notes/summer/4.txt', value: valueDeleteMarker, versionId: 'foo' },
-    { key: 'notes/summer/4.txt', value: valueDeleteMarker, versionId: 'qux' },
-    { key: 'notes/summer/444.txt',
-        value: valueDeleteMarker, versionId: 'null' },
-    { key: 'notes/summer/44444.txt',
-        value: valueDeleteMarker, versionId: 'null' },
+    { key: 'notes/summer/4.txt', value: barDM, versionId: 'bar' },
+    { key: 'notes/summer/4.txt', value: fooDM, versionId: 'foo' },
+    { key: 'notes/summer/4.txt', value: quxDM, versionId: 'qux' },
+    { key: 'notes/summer/444.txt', value: fooDM, versionId: 'foo' },
+    { key: 'notes/summer/44444.txt', value: fooDM, versionId: 'foo' },
```

Comment:
I feel like these 2 arrays can be simplified into only one, as `receivedData` is most likely just using `dataVersioned` array values and adds `versionId` field.. What do you think ?

Reply:
It's actually not an exact 1-1, it depends on the keys, for example you can have a duplicate master/version key, or not (depending if it's V0 or V1 or if it's a current/noncurrent version, or delete marker). Hence I think it's best to keep them separate. There are actually 3 different arrays if you account for the V0 and V1 data.